### PR TITLE
track master patches from return42 fork

### PR DESCRIPTION
**Pull request by return42**
@dalf I add this PR only to track patches from my master branch.  From here you can cherry pick to upstream (if you think it make sense), do not merge this PR, I will continue to send patches.

update: see also gist at https://gist.github.com/dalf/60450e19b042bd39b9ce470c095d89b3

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,4 +1,6 @@
-#!/bin/sh
+#!/usr/bin/env bash
+# -*- coding: utf-8; mode: bash  -*-
+
```
Comment (return42) [inline run.sh:3]: Sorry, that doesn't really belong in this patch. I realized that too late .. the patch was already pushed remote :-o

**Comment by dalf** [comment]
thank you
